install/install.go

```diff
@@ -1052,6 +1053,10 @@ func (p *InstallParameters) validate() error {
 
 		p.configOverwrites[t[0]] = t[1]
 	}
+	var checkVersion = regexp.MustCompile(`(0|[1-9][0-9]*)*\.*(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-[a-zA-Z0-9]+)*\.*(0|[1-9][0-9]*)?`).MatchString
```

Comment:
This regex seems a bit different from the one we had discussed before, in particular there seems to be extra `*` following some of the `\.`, which means that it could accept versions like "0..1.0", which I don't think is intended.

I wonder if there's a better way to get a canonical version regex instead of rolling our own here :thinking: 

On the other hand, if we do need to make our own here, it would be good to create a small unit test to try out various correct and incorrect versions against the regex to check that the regex is linting the input as we would expect.

Reply:
^ On the same train of thought, it would be good to declare the regex only once and use it in multiple places so that this one and the other one don't get out-of-sync (and if we need to fix it again, we can do that once).